### PR TITLE
feat(digitalocean)!: Migrate to Arrow native SDK

### LINKED ISSUE #10739: feat: Migrate `plugin/source/digitalocean` to `github.com/cloudquery/plugin-sdk/v3`
### Which problem is this feature request solving?

Part of #10711

### Describe the solution you'd like

Upgrade the code.

### Pull request (optional)

- [X] I can submit a pull request

### PR DESCRIPTION
**Pull request by disq**
Closes #10739

BEGIN_COMMIT_OVERRIDE
feat: Update to use [Apache Arrow](https://arrow.apache.org/) type system (#10980)

BREAKING-CHANGE: This release introduces an internal change to our type system to use [Apache Arrow](https://arrow.apache.org/). This should not have any visible breaking changes, however due to the size of the change we are introducing it under a major version bump to communicate that it might have some bugs that we weren't able to catch during our internal tests. If you encounter an issue during the upgrade, please submit a [bug report](https://github.com/cloudquery/cloudquery/issues/new/choose). You will also need to update destinations depending on which one you use:
- Azure Blob Storage >= v3.2.0
- BigQuery >= v3.0.0
- ClickHouse >= v3.1.1
- DuckDB >= v1.1.6
- Elasticsearch >= v2.0.0
- File >= v3.2.0
- Firehose >= v2.0.2
- GCS >= v3.2.0
- Gremlin >= v2.1.10
- Kafka >= v3.0.1
- Meilisearch >= v2.0.1
- Microsoft SQL Server >= v4.2.0
- MongoDB >= v2.0.1
- MySQL >= v2.0.2
- Neo4j >= v3.0.0
- PostgreSQL >= v4.2.0
- S3 >= v4.4.0
- Snowflake >= v2.1.1
- SQLite >= v2.2.0

END_COMMIT_OVERRIDE

### CODE REVIEW

**Review by yevgenypats — APPROVED** [review]

**Comment by yevgenypats** [comment]
Actually not sure why it says there are breaking changes. I think it shouldn't maybe something up with the diff tooling

File: plugins/source/digitalocean/go.mod [hunk 1]
```diff
@@ -3,25 +3,24 @@ module github.com/cloudquery/cloudquery/plugins/source/digitalocean
...
 require (
+	github.com/apache/arrow/go/v13 v13.0.0-20230509040948-de6c3cd2b604
 	github.com/avast/retry-go/v4 v4.3.3
 	github.com/aws/aws-sdk-go-v2 v1.18.0
 	github.com/aws/aws-sdk-go-v2/config v1.18.25
 	github.com/aws/aws-sdk-go-v2/service/s3 v1.30.0
 	github.com/aws/smithy-go v1.13.5
 	github.com/cloudquery/plugin-pb-go v1.0.8
-	github.com/cloudquery/plugin-sdk/v2 v2.7.0
+	github.com/cloudquery/plugin-sdk/v3 v3.6.2
```
Comment (candiduslynx) [inline plugins/source/digitalocean/go.mod:13]:
```suggestion
	github.com/cloudquery/plugin-sdk/v3 v3.6.4
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Comment by candiduslynx** [comment]
@disq I've rebased to see if the table diff will be regenerated

File: .github/workflows/release_summary.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ on:
     paths:
       - "plugins/source/**"
       - ".github/workflows/release_summary.yml"
+      - "website/pages/docs/plugins/sources/*/tables/*"
```
Comment (candiduslynx) [inline .github/workflows/release_summary.yml:8]:
```suggestion
```

File: plugins/source/digitalocean/resources/services/monitoring/alert_policies.go [hunk 3]
```diff
@@ -14,12 +15,10 @@ func AlertPolicies() *schema.Table {
...
 			{
-				Name:     "uuid",
-				Type:     schema.TypeString,
-				Resolver: schema.PathResolver("UUID"),
-				CreationOptions: schema.ColumnCreationOptions{
-					PrimaryKey: true,
-				},
+				Name:       "uuid",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   schema.PathResolver("UUID"),
```
Comment (candiduslynx) [inline plugins/source/digitalocean/resources/services/monitoring/alert_policies.go:20]: shouldn't it be UUID?

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**